src/utils.ml

```diff
@@ -1,5 +1,29 @@
 open Import
 
+
+let get_first_lines ic prefix_read =
+  let prefix_read = prefix_read ^ input_line ic in
+  let first_lines = Array.make 3 "" in
+  let lines_list = String.split_on_char prefix_read ~sep:'\n' in
+  let number_read_lines = List.length lines_list in
+  for i = 0 to number_read_lines - 1 do
+    first_lines.(i) <- List.nth lines_list i
+  done;
+  for i = number_read_lines to 2 do
+    first_lines.(i) <- input_line ic
+  done;
+  first_lines
+
+let skip_hash_bang ic prefix_read =
+  if
+    String.length prefix_read > 1
+    && String.equal (String.sub prefix_read ~pos:0 ~len:2) "#!"
+  then
+    let first_lines = get_first_lines ic prefix_read in
+    if String.equal first_lines.(2) "!#" then ""
+    else first_lines.(1) ^ first_lines.(2)
```

Comment:
I think we are missing a `\n` between the two lines.

Reply:
Since I've followed your idea to rewrite the function using ocamllex, I'll close this conversation.